internal/apiserver/route_get_identity_by_id.go

```diff
@@ -41,10 +43,30 @@ var getIdentityByID = &ffapi.Route{
 	JSONOutputCodes: []int{http.StatusOK},
 	Extensions: &coreExtensions{
 		CoreJSONHandler: func(r *ffapi.APIRequest, cr *coreRequest) (output interface{}, err error) {
-			if strings.EqualFold(r.QP["fetchverifiers"], "true") {
-				return cr.or.NetworkMap().GetIdentityByIDWithVerifiers(cr.ctx, r.PP["iid"])
+			switch id := r.PP["id"]; {
+			case isUUID(id):
+				if strings.EqualFold(r.QP["fetchverifiers"], "true") {
+					return cr.or.NetworkMap().GetIdentityByIDWithVerifiers(cr.ctx, id)
+				}
+				return cr.or.NetworkMap().GetIdentityByID(cr.ctx, id)
+			case isDID(id):
+				if strings.EqualFold(r.QP["fetchverifiers"], "true") {
+					return cr.or.NetworkMap().GetIdentityByDIDWithVerifiers(cr.ctx, id)
+				}
+				return cr.or.NetworkMap().GetIdentityByDID(cr.ctx, id)
 			}
-			return cr.or.NetworkMap().GetIdentityByID(cr.ctx, r.PP["iid"])
+			return nil, i18n.NewError(cr.ctx, i18n.MsgUnknownIdentityType)
 		},
 	},
 }
+
+// isUUID checks if the given string is a UUID
+func isUUID(s string) bool {
+	_, err := uuid.Parse(s)
+	return err == nil
+}
+
+// isDID checks if the given string is a potential DID
+func isDID(s string) bool {
+	return strings.HasPrefix(s, "did:")
+}
```

Comment:
Keep the same level of check and change this to a regex expression - I think we should have one for DIDs already in the codebase

Reply:
Since we have the DID validation happening later in the code, I preferred to determine if this a DID string or not. Because if this is a DID but a not valid one, I want the user to receive the proper error code (vs in this case the error code will be "unknown format"). wdyt?

Reply:
ah because you are not catching the error in the switch statement, you could catch the error there if you wish to? It just feels since you have moved the validation here make sense to construct the correct error instead of relying on the function underneath to throw the right thing - but will leave up to you